specs/altair/sync-protocol.md

```diff
@@ -102,6 +103,15 @@ def get_subtree_index(generalized_index: GeneralizedIndex) -> uint64:
     return uint64(generalized_index % 2**(floorlog2(generalized_index)))
 ```
 
+### `fork_version_at_epoch`
+
+```python
+def fork_version_at_epoch(epoch: Epoch) -> Version:
+    if epoch >= ALTAIR_FORK_EPOCH:
+        return ALTAIR_FORK_VERSION
+    return GENESIS_FORK_VERSION
+```
```

Comment:
One problem here comes up during testing `@with_altair_and_later`, because the generated Altair states do not fork at `ALTAIR_FORK_EPOCH` etc but instead fork earlier. So the local `state` during test is already on Altair, but the validation fails on the client side because it assumes that the fork applies after `ALTAIR_FORK_EPOCH`.

The previous approach had a similar problem that the light client would just blindly trust whatever `fork_version` is given as part of the `LightClientUpdate`.

What is the recommended way to continue here?

Reply:
In a naive case the validator could try with verifying the signature with both sync committees

Reply:
The `fork_version` could theoretically change multiple times throughout a single sync committee period. 
But yeah, those could also be locally enumerated and tested.
Problem remains, though. The client needs a way to verify / determine a fork version, or it runs risk of accepting an update with an invalid fork version.

Reply:
Having the `fork_version` included actually enables use cases where the light client does not need to be aware of forks, as long as it does not intend to request state proofs and continue operating on the `BeaconBlockHeader`. An example could be a simple tool to advance a given checkpoint root to the most recent `BeaconBlockHeader`, then fetches that state from Infura and uses it to start a full node using weak subjectivity sync.

For now, I'll keep the `fork_version` in. It should be revisited in a separate PR if needed.